mlflow/tracking/client.py

```diff
@@ -218,15 +220,19 @@ def log_batch(self, run_id, metrics=(), params=(), tags=()):
 
     def log_artifact(self, run_id, local_path, artifact_path=None):
         """
-        Write a local file to the remote ``artifact_uri``.
+        Write a local file or directory to the remote ``artifact_uri``.
```

Comment:
After testing, I've confirmed that `mlflow.log_artifacts` has the same semantics as `mlflow.log_artifact` for a directory input: MLflow logs the *contents* of the directory; it does not create the outer directory in the artifact store.

E.g., consider the following directory structure:
```
    outer_dir
    ├── inner_dir
    |   ├── inner_file.txt
    ├── file.txt
```

Calling `mlflow.log_artifact(s)("output_dir")` produces the following artifact contents:
```
    ├── inner_dir
    |   ├── inner_file.txt
    ├── file.txt
```
`outer_dir` is omitted.

Can we make this behavior clear in the documentation?

Reply:
I actually think it's more sensible to create the outer directory. Logging a file makes that file visible as an artifact, so logging a directory should make that directory visible as is. 
@smurching @mateiz  thoughts?